.github/workflows/hyperfine.yml

```diff
@@ -0,0 +1,199 @@
+name: Hyperfine Benchmark
+
+on:
+  pull_request:
+    branches: [ '*' ]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build-programs:
+    strategy:
+      matrix:
+        branch: [ base, head ]
+    name: Build Cairo programs for ${{ matrix.branch }}
+    runs-on: ubuntu-22.04
+    outputs:
+      benchmark-hashes-base: ${{ steps.export-hashes.outputs.benchmark-hashes-base }}
+      benchmark-hashes-head: ${{ steps.export-hashes.outputs.benchmark-hashes-head }}
+    steps:
+      - name: Checkout base commit
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request[matrix.branch].sha }}
+
+      - name: Fetch from cache
+        uses: actions/cache@v3
+        id: cache
+        with:
+          path: base_programs/*.json
+          key: benchmarks-${{ hashFiles( 'cairo_programs/benchmarks/*.cairo' ) }}
+          restore-keys: benchmarks-
```

Comment:
Wouldn't this, combined with skipping the steps of contract compilation on cache hit, produce errors when one of them changed?